src/Raven.Client/Documents/Operations/Attachments/AttachmentsStream.cs

```diff
@@ -0,0 +1,367 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Operations.Attachments
+{
+    internal class AttachmentsStreamInfo
+    {
+        public AttachmentsStreamInfo()
+        {
+            AllStreams = new Dictionary<string, MemoryStream>();
+            CurrentPosition = 0;
+            CurrentIndex = 0;
+        }
+
+        public AttachmentsDetails AttachmentAdvancedDetails;
+        public readonly Dictionary<string, MemoryStream> AllStreams;
+        public JsonOperationContext.ManagedPinnedBuffer Buffer;
+
+        public int CurrentIndex;
+        public long CurrentPosition;
+        public bool BufferExposed;
+    }
+
+    internal class AttachmentsStream : Stream
+    {
+        private Stream _baseStream;
+        private readonly string _name;
+        private readonly AttachmentsStreamInfo _info;
+        private readonly int _index;
+        private readonly int _size;
+        private bool _merged;
+
+        public AttachmentsStream(Stream baseStream, string name, int index, int size, AttachmentsStreamInfo info)
+        {
+            if (baseStream == null)
+                throw new ArgumentNullException(nameof(baseStream));
+            if (baseStream.CanRead == false)
+                throw new ArgumentException("can't read base stream");
+
+            _baseStream = baseStream;
+            _name = name;
+            _info = info;
+            _index = index;
+            _size = size;
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            CheckDisposed();
+
+            if (buffer == null)
+                throw new ArgumentNullException($"buffer is null.");
+
+            if (offset + Math.Min(_size, count) > buffer.Length)
```

Comment:
You can calculate it once (Math.Min(_size, count))

Reply:
done